crates/shared/src/current_block.rs

```diff
@@ -124,13 +140,32 @@ where
     }
 }
 
+/// Only updates the current block if the new block number is strictly bigger than the current one.
+/// Emits error logs when the new block number differs by more than `MAX_REORG_BLOCK_COUNT` from
+/// the current block number.
```

Comment:
If we do this then let's update the comment at line 20 in this file. It currently says that we send on the stream whenever the current block changes. With this change it would be more accurate to say that we signal current block increasing. It is a bit different.
And then I'd double check that everything using the block stream really wants the new behavior. In some cases I feel our goal was to update components to what we believe to be the current chain state. Does it not make sense to still do that? I guess we are kind of saying that those changes are rare/node problems and can be ignored but it feels a bit weird.

Reply:
That's a good point. I will investigate which components depend or expect the old behavior and the update the PR accordingly.

Reply:
I did some research how the existing components react to blocks in the past and whether they would profit from the change in the block stream.
`SolvableOrdersCache`: Doesn't get notified and only observes current block, writes that block into the auction and uses block number to determine if the cached user balances can be reused.
`Driver`: Also only observes latest block and fetches liquidity from that block. This could technically lead to a problem when the driver queries a block that the current node is not aware of but I would argue that this is okay because the alternative is sending very outdated liquidity with the auction.
`Orderbook`: Also only observes the latest block and determines whether the current auction is up-to-date. If the latest block of the order book is too much in the past compared to the block in the auction the auction gets discarded and the `/auction` returns 404.
`RecentBlockCache` (used for balancer and uniswap pool updates): Updates triggered through maintenance (initiated by the current block stream yielding a new block). This component wouldn't have a big problem with blocks in the past. There is a code path that would yield old data when you query the cache but that is never taken.

Looks like the PR would be an improvement for all the components. So I just adapted the comment you suggested and would call it a day.